cmd/cogito/main.go

```diff
@@ -18,15 +20,30 @@ func main() {
 	// - stdin, stdout and command-line arguments for the protocol itself
 	// - stderr for logging
 	// See: https://concourse-ci.org/implementing-resource-types.html
-	if err := run(os.Stdin, os.Stdout, os.Stderr, os.Args); err != nil {
-		fmt.Fprintf(os.Stderr, "Error: %s\n", err)
+	if err := mainErr(os.Stdin, os.Stdout, os.Stderr, os.Args); err != nil {
+		fmt.Fprintf(os.Stderr, "cogito: error: %s\n", err)
 		os.Exit(1)
 	}
 }
 
-func run(in io.Reader, out io.Writer, logOut io.Writer, args []string) error {
+func mainErr(in io.Reader, out io.Writer, logOut io.Writer, args []string) error {
+	cmd := path.Base(args[0])
+	validCmds := sets.From("check", "in", "out")
+	if !validCmds.Contains(cmd) {
+		return fmt.Errorf("invoked as '%s'; want: one of %v", cmd, validCmds)
+	}
+
+	input, err := io.ReadAll(in)
+	if err != nil {
+		return fmt.Errorf("reading stdin: %s", err)
+	}
+	logLevel, err := peekLogLevel(input)
```

Comment:
The main idea is here: peek at the log_level from main.
To be able to then parse the input again, I also had to read the stdin stream once into a buffer and pass it around.